lib/src/position.dart

```diff
@@ -374,6 +376,79 @@ abstract class Position<T extends Position<T>> {
     }
   }
 
+  /// Create a [Position] from setup and variants
+  static Position setupPosition(Variant rules, Setup setup,
+      {bool? ignoreImpossibleCheck}) {
+// TODO:missing horde, racingkings. Returns Chess for those variants
+    switch (rules) {
+      case Variant.chess:
+        return Chess.fromSetup(setup,
+            ignoreImpossibleCheck: ignoreImpossibleCheck);
+      case Variant.antichess:
+        return Antichess.fromSetup(setup,
+            ignoreImpossibleCheck: ignoreImpossibleCheck);
+      case Variant.atomic:
+        return Atomic.fromSetup(setup,
+            ignoreImpossibleCheck: ignoreImpossibleCheck);
+      case Variant.kingofthehill:
+        return KingOfTheHill.fromSetup(setup,
+            ignoreImpossibleCheck: ignoreImpossibleCheck);
+      case Variant.crazyhouse:
+        return Crazyhouse.fromSetup(setup,
+            ignoreImpossibleCheck: ignoreImpossibleCheck);
+      case Variant.threecheck:
+        return ThreeCheck.fromSetup(setup,
+            ignoreImpossibleCheck: ignoreImpossibleCheck);
+      default:
```

Comment:
let's cover all the `case` and throw `UnimplementedError()` for missing variants instead